tests/test_spstats.py

```diff
@@ -171,3 +170,49 @@ def test_patches_method(self):
             gsd=diff.res[0],
             area_size=10000
         )
+
+class TestBinning:
+
+    def test_nd_binning(self):
+
+        ref, diff, mask = load_ref_and_diff()
+
+        slope, aspect = xdem.coreg.calculate_slope_and_aspect(ref.data.squeeze())
+
+        # 1d binning, by default will create 10 bins
+        df = xdem.spstats.nd_binning_scipy(dh=diff.data.flatten(),list_var=[slope.flatten()],list_var_names=['slope'])
+
+        # check length matches
+        assert df.shape[0] == 10
+        # check bin edges match the minimum and maximum of binning variable
+        assert np.nanmin(slope) == np.min(pd.IntervalIndex(df.slope).left)
+        assert np.nanmax(slope) == np.max(pd.IntervalIndex(df.slope).right)
+
+        # 1d binning with 20 bins
+        df = xdem.spstats.nd_binning_scipy(dh=diff.data.flatten(), list_var=[slope.flatten()], list_var_names=['slope'],
+                                           list_var_bins=[[20]])
+        # check length matches
+        assert df.shape[0] == 20
+
+        # nmad goes up quite a bit with slope, we can expect a 10 m measurement error difference
+        assert df.nmad.values[-1] - df.nmad.values[0] > 10
+
+        # try custom stat
+        def percentile_80(a):
+            return np.nanpercentile(a, 80)
+
+        # check the function runs with custom functions
+        xdem.spstats.nd_binning_scipy(dh=diff.data.flatten(),list_var=[slope.flatten()],list_var_names=['slope'], statistics=['count',percentile_80])
```

Comment:
I assume that `count` is a statistic that is built into scipy? What happens if I want the statistic `blasjdhakwjhd`? Does it fail and raise a descriptive error? That would be nice to test!

Reply:
scipy raises an error indeed: `ValueError: invalid statistic blasjdhakwjhd`
This is checked within their tests, I don't think we need to add something ourselves when it's simply passing a parameter directly? Otherwise the tests would become very long.